File: Java, PHP, JSP, Calculator/calc.java
// Jeremiah S Kleiber Calculator in Java 10/12/20 assignment 2 ISP

import java.util.Scanner;  // Import the Scanner class
import java.util.InputMismatchException;
import java.util.* ;

public class calc {
 	public static void main(String[] args) 
	{
		menu();// calls menu to start
	
  	}
	//menu displays all options, call memeber functions, and validates
	static void menu()
	{
		Scanner inObj = new Scanner(System.in);  // Create a Scanner object
		System.out.println("Calculator in java");
		System.out.println("Enter + to add");
		System.out.println("Enter - to subtract");
		System.out.println("Enter * to multiply");
		System.out.println("Enter / to divide");
		System.out.println("Enter % to find remainder");
		System.out.println("Enter q to quit");
		String menuOption = inObj.nextLine(); 
		
		switch(menuOption)// checks input
		{
			case "+": // adding
			{
				System.out.println("Enter 1st number");
				float val1 = 0;
				float val2 = 0;

				if(inObj.hasNextInt() || inObj.hasNextFloat()) // validation
				{ 
					val1 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
			
				}

				System.out.println("Enter 2nd number");

				if(inObj.hasNextInt() || inObj.hasNextFloat()) // validation
				{ 
					val2 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
				}

				add(val1,val2);
				menu();
				
				break;
			}
			case "-": //subtract
			{
				System.out.println("Enter 1st number");
				float val1 = 0;
				float val2 = 0;

				if(inObj.hasNextInt() || inObj.hasNextFloat())// validation
				{ 
					val1 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
				}

				System.out.println("Enter 2nd number");

				if(inObj.hasNextInt() || inObj.hasNextFloat())// validation
				{ 
					val2 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
				}

				sub(val1,val2);
				menu();

				break;
			}
			case "*": // multiply
			{
				System.out.println("Enter 1st number");
				float val1 = 0;
				float val2 = 0;

				if(inObj.hasNextInt() || inObj.hasNextFloat())// validation
				{ 
					val1 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
			
				}

				System.out.println("Enter 2nd number");
				if(inObj.hasNextInt() || inObj.hasNextFloat())// validation
				{ 
					val2 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
				}

				mult(val1,val2);
				menu();

				break;
			}
			case "/": // divide
			{
				System.out.println("Enter 1st number");
				float val1 = 0;
				float val2 = 0;

				if(inObj.hasNextInt() || inObj.hasNextFloat())// validation
				{ 
					val1 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
			
				}

				System.out.println("Enter 2nd number");

				if(inObj.hasNextInt() || inObj.hasNextFloat())// validation
				{ 
					val2 = inObj.nextFloat();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
				}

				divide(val1,val2);
				menu();

				break;
			}
			case "%": // remainder
			{
				System.out.println("Enter 1st number");
				int val1 = 0;
				int val2 = 0;

				if(inObj.hasNextInt() )// validation
				{ 
					val1 = inObj.nextInt();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
				}

				System.out.println("Enter 2nd number");

				if(inObj.hasNextInt() )// validation
				{ 
					val2 = inObj.nextInt();
				}
				else
				{
					System.out.println("Incorrect entry, returning to menu");
					menu();
					return;
				}

				remainder(val1,val2);
				menu();

				break;
			}
			case "q": // exits
			{
				System.out.println("Ending program");
				break;
			}
			default: // validation for menu
			{
				System.out.println("Incorrect value entered, only enter +,-,*,/,%,q");
				menu();
			}
		}
	}
	// adds 2 values and returns them
	static void add(float val1, float val2) 
	{
		float val3 = val1 + val2;
		System.out.println(val3);
	}
	//subtract 2 values and returns them
	static void sub(float val1, float val2) 
	{
		float val3 = val1 - val2;
		System.out.println(val3);
	}
	//multiply 2 values and returns them
	static void mult(float val1, float val2) 
	{
		float val3 = val1 * val2;
		System.out.println(val3);
	}
	//divides 2 values and returns them
	static void divide(float val1, float val2) 
	{
		if(val2 == 0)// extra step in validation
		{
			System.out.println("Can't divide by 0, setting it to one");
			val2 = 1;	
		}
		float val3 = val1 / val2;
		System.out.println(val3);
	}
	//finds the remainder of 2 values and returns them
	static void remainder(int val1, int val2) 
	{
		if(val2 == 0)// extra step in validation
		{
			System.out.println("Can't divide by 0, setting it to one");
			val2 = 1;	
		}
		int val3 = val1 % val2;
		System.out.println(val3);
	}

}

File: Java, PHP, JSP, Calculator/calc.php
<!--Calculator using php. Jeremiah S Kleiber 10/15/20 jsk84@zips.uakron.edu assignment 2-->
<!DOCTYPE html>
<html>

<head><title>Calculator</title></head>
<body><h1>A basic calculator</h1>
<form method=POST>Please enter numbers here: 
<input name=number1>
<input name=number2>
<br/>
	<input type="submit" name="+" value="+" />
	<input type="submit" name="-" value="-" />
	<input type="submit" name="*" value="*" />
	<input type="submit" name="/" value="/" />
	<input type="submit" name="%" value="%" />
</form>


<?php

#setting the user entries into a variable
$val1 = $_POST["number1"];
$val2 = $_POST["number2"]; 

if($_POST["number1"] != null && $_POST["number2"] != null) # seeing if the user entered 2 values
{
	# testing or validating the values to make sure they are an int other wise program wont run
	if ( (filter_var($val1, FILTER_VALIDATE_INT) === 0 || filter_var($val1, FILTER_VALIDATE_INT))
		&& (filter_var($val2, FILTER_VALIDATE_INT) === 0 || filter_var($val2, FILTER_VALIDATE_INT)) )  
	{
		echo "<br>";
		
		if($_POST["+"] != null) # adding
		{
			echo $val1 + $val2;
		}
		else if($_POST["-"] != null) # subtracting
		{
			echo $val1 - $val2;
		}
		else if($_POST["*"] != null) # mult
		{
			echo $val1 * $val2;
		}
		else if($_POST["/"] != null) # division
		{
			if($val2 == 0)
			{
				echo "Can't Divide by zero, setting val2 equal to 1, : ";
				$val2 = 1;
			}
			echo $val1 / $val2;
		}
		else if($_POST["%"] != null) # remainder
		{
			if($val2 == 0)
			{
				echo "Can't Divide by zero, setting val2 equal to 1. : ";
				$val2 = 1;
			}
			if($val1 < $val2)
			{
				echo "Can't find the remainder of a fraction, setting first number to the second. : ";
				$val1 = $val2;
			}
			echo $val1 % $val2;
		}
	}
	else # if one or more values wasnt an interger
	{
		echo "<br>";
		echo("Variable is not an integer");
	}
}
else if($_POST["number1"] != null || $_POST["number2"] != null) # if the user only entered 1 value
{
	echo "<br>";
	echo "Only one value was entered";
}
echo "<br>";


?>
<br/>
<p1> Made by Jeremiah S Kleiber </p1>
</body>
</html>